mesonbuild/interpreter/interpreter.py

```diff
@@ -1620,7 +1620,7 @@ def func_subdir_done(self, node, args, kwargs):
                       'build_always', 'capture', 'depends', 'depend_files', 'depfile',
                       'build_by_default', 'build_always_stale', 'console', 'env',
                       'feed', 'install_tag'})
-    @typed_pos_args('custom_target', str)
+    @typed_pos_args('custom_target', optargs=[str])
     def func_custom_target(self, node: mparser.FunctionNode, args: T.Tuple[str], kwargs: 'TYPE_kwargs') -> build.CustomTarget:
```

Comment:
I'm not sure how I feel about making the name optional, it effectively means we can't ever add any required positional arguments to custom target without breaking backwards compatibility. However, you haven't updated the signature here, it needs to be Optional[str]

Reply:
We could add other args, just that if you want to pass more than one arg then the name becomes required. You could still do `custom_target('', other_arg)`. A bit ugly but would work in the very unlikely event this ever happen.

Reply:
> you haven't updated the signature here, it needs to be Optional[str]

done now.